### DISCUSSION TITLE
Support of .Net 7.0

### DISCUSSION THREAD

**Discussion by maulikshah88** [PiranhaCMS/piranha.core#1992] [General]
Hi,

Is there any release plan to migrate on .net 7.0 ?

**Comment by tidyui** [comment]
Hi there! We have updated all of the dependencies on Piranha to make sure that you can include Piranha in a `.NET7` application, however we never recompile the framework itself for Microsofts short term releases. We usually wait until the next LTS (which in this case is `.NET8`) to upgrade everything as it is a breaking change.

Regards

**Comment by selaromdotnet** [comment]
hi, could you elaborate a bit on this? I have a .net 7 project I'd love to integrate with PiranhaCMS but I'm not sure if I can. I was able to add the nuget packages that are .net 6 and it built, but it wouldn't run...

It sounds like I need to pull the source and build it, and reference the .net 7 DLLs that it creates, manually adding them to my project instead of using nuget. 

is that correct, will it work? I'm gonna try it anyway, but curious to hear more about this, as I don't want to have to wait for .net 8 to use this, nor do i want to downgrade my solution to ,.net6

**Comment by selaromdotnet** [comment]
holy cow that totally worked! I was able to build the solution for .net7, added all the references to my solution and now I have a CMS backend for my blazor server app!!

now I can manage the copy, images, etc of my static pages in a CMS instead of on the blazor pages, this is FANTASTIC!!!

If anyone's interested in how I did this, lemme know and I'll write up a blog post about it, thank you so much for this project!!!
